php/php-rr-sdk.md

```diff
@@ -0,0 +1,692 @@
+    return new Response('something');
+}
+```
+
+It is mandatory to use proper type declarations on input and return types of activity. While PHP 7.4 is not widely adopted,
+it is recommended to use strict property typing for payload properties as well when possible or highlight the type via
+mapper specific annotation.
+
+### Process Isolation
+Activity handler will stay in memory permanently, only the execution context will change. Such an approach will ellimitate 
+the framework initialization overhead and reduce memory consumption. 
+
+The system must limit to only one concurrent execution per worker, which makes possible to use all existing PHP SPL and vendor
+libraries without changes. 
+
+## Workflows
+Unlike activities, the workflow execution must be based on a physical memory object. In order to ensure proper density
+and effectiveness of the system, multiple workflows will locate inside singe PHP process. Each workflow will operate
+as one or multiple [Generators](https://www.php.net/manual/en/class.generator.php). The execution can be controlled using
+system [keyword](https://www.php.net/manual/en/language.generators.syntax.php) `yield`.
+
+Yield keyword should be used to request the command execution by Temporal system. For the sake of simplify each command
+must be presented as typed object, though, end user API can encapsulate object construction using helper functions. 
+
+Each yield command will return Promise/Future object which can later be referenced in a code. Simple workflow execution might
+look as following:
+
+```php
+use Temporal\Workflow;
+
+class UploadWorkflow
+{
+    public function run(Workflow\ContextInterface $ctx, string $input): string
+    {
+        $promise = yield new Workflow\ExecuteActivy(
+            'activityName',
+            $input, 
+            100 // startToFinishTimeout, etc. 
+        );        
+    
+        $result = yield $promise->get(Obj::class);
+
+        return $result->getValue();
+    }
+}
+```
+
+Read more about proposed syntax approach and syntax sugar down below. The given example can be simplified into:
+
+```php
+use Temporal\Workflow;
+
+class UploadWorkflow extends Workflow\Workflow
+{
+    public function run(string $input): string
+    {
+        $result = yield $this->activities
+            ->withStartToFinish(100)
+            ->activityName($input)
+            ->get(Obj::class);
+
+        return $result->getValue();
+    }
+}
+```
+
+> No common interface required.
+
+### Registration
+Similar to activities each workflow type must be registered within worker context, since worklow might expose more that 
+one method (query, primary method) meta declation is required. 
+
+```php
+$worker = new WorkflowWorker();
+
+$workflow = (new WorkflowDeclaration('name'))
+                ->setClass(WofkflowClass::class)
+                ->setHandlerMethod('run')
+                ->addQueryHandler('name', 'method-name')
+                ->addSignalHandler('name', 'method-name');
```

Comment:
This looks weird as which methods are which is property of the workflow and not how it is registered. What about doing dynamic registration of query and signal handlers from the workflow code?

Reply:
Dynamic registration is possible as well. This is just an example of explicit registration outside of the workflow. This is needed since PHP (at the moment) does not provide native support for annotations and attributes, so _something_ needs to parse and provide them to the Temporal engine. 

Consider this as deep internal logic which most users will never see. It's only needed to make the actual workflow code to as bootstrap-less and verbose as possible.